src/zProposal.ts

```diff
@@ -0,0 +1,250 @@
+import { TransactionResponse } from '@ethersproject/abstract-provider';
+import { ethers } from 'ethers';
+
+import GnosisSafeClient from './gnosis-safe';
+import SnapshotClient from './snapshot-io';
+import {
+  Choice,
+  Proposal,
+  ProposalId,
+  TokenMetaData,
+  Vote,
+  VoteChoice,
+} from './types';
+import { t } from './utilities/messages';
+import zDAOClient from './zDAOClient';
+
+class zProposal implements Proposal {
+  private readonly _zDAO: zDAOClient;
+  private readonly _snapshotClient: SnapshotClient;
+  private readonly _gnosisSafeClient: GnosisSafeClient;
+  private readonly _id: ProposalId;
+  private readonly _type: string;
+  private readonly _author: string;
+  private readonly _title: string;
+  private readonly _body: string;
+  private readonly _ipfs: string;
+  private readonly _choices: VoteChoice[];
+  private readonly _created: Date;
+  private readonly _start: Date;
+  private readonly _end: Date;
+  private readonly _state: 'pending' | 'active' | 'closed';
+  private readonly _network: string;
+  private readonly _snapshot: string;
+  private readonly _scores: number[];
+  private readonly _votes: number;
+  private _metadata: TokenMetaData | undefined;
+
+  constructor(
+    zDAO: zDAOClient,
+    snapshotClient: SnapshotClient,
+    gnosisSafeClient: GnosisSafeClient,
+    id: ProposalId,
+    type: string,
+    author: string,
+    title: string,
+    body = '',
```

Comment:
This is nullable in `Proposal` but is defaulted to `''` here. Can we remove the nullable on the interface?